### PR TITLE
Improve Daylight Shavings Helmet tile

### PR DESCRIPTION
**Pull request by ajcoppa**
Apologies for the horrible diff.

Note: I don't actually have a DSH, so I can't test every piece of the logic. I did do a lot of manual testing by directly setting the class and buff order to certain values and everything looked good. I'd love someone with an actual helmet to help make sure everything really does work.

- Fixed a bug that was breaking handling for different classes (formula was using + instead of *)
- Buff table now shows upcoming buffs based on your class
- Removed some code duplication

### CODE REVIEW

File: Source/relay/TourGuide/Items of the Month/2021/Daylight Shavings Helmet.ash [hunk 1]
```diff
@@ -1,93 +1,90 @@
 //Daylight Shavings Helmet
+
+string [int, string] dshBuffOrder = {
```
Comment (docrostov) [inline Source/relay/TourGuide/Items of the Month/2021/Daylight Shavings Helmet.ash:3]: this is a slick and markedly better way to achieve the result of the tooltip table below. good work.

File: Source/relay/TourGuide/Items of the Month/2021/Daylight Shavings Helmet.ash [hunk 2]
```diff
@@ -1,93 +1,90 @@
...
+string [string] getNthClassBuff(int n) {
+	int classInt = my_class().to_int();
+	int buffIndex;
+	if (classInt < 7)
+	{
+		buffIndex = ((classInt * n) % 11);
+	}
+	else if (classInt > 6)
+	{
+		buffIndex = ((((classInt % 6 + 1) % 11)));
```
Comment (docrostov) [inline Source/relay/TourGuide/Items of the Month/2021/Daylight Shavings Helmet.ash:26]: hrm wasn't there a thing where boris had some weird exception handling? or is this the actual fix... i haven't actually worried about DSH much in avatar paths that didn't have pins telling me exactly what the dsh alignment was lmao

File: Source/relay/TourGuide/Items of the Month/2021/Daylight Shavings Helmet.ash [hunk 3]
```diff
@@ -1,93 +1,90 @@
...
+	else {
+		shavingBuffCounter = beardBuff - 2665;
+	}
+
+	// Shaving buff prediction
+	string [string] nextBeardBuff = getNthClassBuff(shavingBuffCounter + 1);
+
+	string nextBeardBuffDescription;
+	string nextBeardBuffEffect;
+	foreach nextBeardBuffName in nextBeardBuff {
```
Comment (docrostov) [inline Source/relay/TourGuide/Items of the Month/2021/Daylight Shavings Helmet.ash:56]: i know this isn't enormously complicated but big props on making this code much more concise, snappy, and straightforward. i appreciate the lil generators and stuff you've got here, it works quite well.

**Review by docrostov — APPROVED** [review]
ok i am going to approve and merge this tomorrow morning so i can test it in my run. one small last validation step, if you want to do so... i -think- CHiT actually has a proper handling of this on a class-by-class basis, may be worth checking CHIT's code?

File: Source/relay/TourGuide/Items of the Month/2021/Daylight Shavings Helmet.ash (hunk 2 above)
Comment (ajcoppa) [reply Source/relay/TourGuide/Items of the Month/2021/Daylight Shavings Helmet.ash:26]: Good question. I didn't change anything about the avatar path logic that was already there, but the formula here matches up with what was in the spading spreadsheet, so I believe it should work.

File: Source/relay/TourGuide/Items of the Month/2021/Daylight Shavings Helmet.ash (hunk 1 above)
Comment (ajcoppa) [reply Source/relay/TourGuide/Items of the Month/2021/Daylight Shavings Helmet.ash:3]: 👍 Doing this in ASH is a little weird compared to TS, but I really wanted to get rid of the duplication, switch/case, etc and this direction ended up working out okay

**Pull request merged**